### PR TITLE
Added functionality to flag a chore as completed on the backend and added associated endpoint

### LINKED ISSUE #239: Add functionality to flag a chore as completed on the backend and write associated endpoint
**User story**
As a developer, I want to be able to flag a chore as completed on the backend in response to an API call from the frontend.

**Acceptance criteria**
 - Implement endpoint per API description on wiki
 - Change value of 'completed' in chore repository
 - Write appropriate unit tests

**Additional context**
https://github.com/SOFTENG701G1/Flatmate-Management-System/wiki/Chores-API

### PR DESCRIPTION
**Pull request by tant836**
… backend

Added functionality to flag a chore as completed on the backend and added associated endpoint. And added tests

## Pull Request Information
Closes #239

**What is the PR Type:** 

**Describe the added/fixed behaviour:**

## Checklist

- [x] Pull request has suitable title and labels, adequately described and associated with a github issue
- [x] Branch name follows correct convention (e.g. feature/#[id]-some-feature)
- [x] Branch builds and executes with no errors
- [x] Test cases has been generated for the feature and test cases pass
- [ ] Updated relevant wiki/documentation

### CODE REVIEW

File: WebApiBackend/WebApiBackend/Controllers/ChoreController.cs [hunk 1]
```diff
@@ -151,5 +151,37 @@ where u.FlatId.Equals(flatId)
...
+            //check chore exists
+            if (chore == null)
+            {
+                return BadRequest();
+            }
+
+            chore.Completed = true;
+            await _choresRepository.Add(chore);
+
+            return NoContent();
```
Comment (lken274) [inline WebApiBackend/WebApiBackend/Controllers/ChoreController.cs:183]: please make this 'Ok'

File: WebApiBackend/WebApiBackend/Controllers/ChoreController.cs [hunk 2]
```diff
@@ -151,5 +151,37 @@ where u.FlatId.Equals(flatId)
...
+            //check identity
+            ClaimsIdentity identity = HttpContext.User.Identity as ClaimsIdentity;
+            int userID = Int16.Parse(identity.FindFirst(ClaimTypes.NameIdentifier).Value);
+            var user = await _userRepository.Get(userID);
+            if (user == null)
+            {
+                return new BadRequestResult();
+            }
+
+            Chore chore = await _choresRepository.Delete(choreId);
```
Comment (lken274) [inline WebApiBackend/WebApiBackend/Controllers/ChoreController.cs:173]: You should be able to flip the value without deleting the chore.

Comment (tant836) [reply WebApiBackend/WebApiBackend/Controllers/ChoreController.cs]: Done

File: WebApiBackend/WebApiBackend/Controllers/ChoreController.cs (hunk 1 above)
Comment (tant836) [reply WebApiBackend/WebApiBackend/Controllers/ChoreController.cs]: Done

**Pull request merged**